clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/IngestionMetadataRetrieverDelegator.java

```diff
@@ -0,0 +1,68 @@
+package com.linkedin.davinci.storage;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.linkedin.davinci.kafka.consumer.KafkaStoreIngestionService;
+import com.linkedin.davinci.kafka.consumer.ReplicaHeartbeatInfo;
+import com.linkedin.davinci.listener.response.AdminResponse;
+import com.linkedin.davinci.listener.response.TopicPartitionIngestionContextResponse;
+import com.linkedin.davinci.stats.ingestion.heartbeat.HeartbeatMonitoringService;
+import com.linkedin.venice.helix.VeniceJsonSerializer;
+import com.linkedin.venice.utils.ComplementSet;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class IngestionMetadataRetrieverDelegator implements IngestionMetadataRetriever {
+  private static final Logger LOGGER = LogManager.getLogger(IngestionMetadataRetrieverDelegator.class);
+  private final KafkaStoreIngestionService kafkaStoreIngestionService;
+  private final HeartbeatMonitoringService heartbeatMonitoringService;
+
+  private final VeniceJsonSerializer<Map<String, ReplicaHeartbeatInfo>> replicaInfoJsonSerializer =
+      new VeniceJsonSerializer<>(new TypeReference<Map<String, ReplicaHeartbeatInfo>>() {
+      });
+
+  public IngestionMetadataRetrieverDelegator(
+      KafkaStoreIngestionService kafkaStoreIngestionService,
+      HeartbeatMonitoringService heartbeatMonitoringService) {
+    this.kafkaStoreIngestionService = kafkaStoreIngestionService;
+    this.heartbeatMonitoringService = heartbeatMonitoringService;
+  }
+
+  @Override
+  public ByteBuffer getStoreVersionCompressionDictionary(String topicName) {
+    return kafkaStoreIngestionService.getStoreVersionCompressionDictionary(topicName);
+  }
+
+  @Override
+  public AdminResponse getConsumptionSnapshots(String topicName, ComplementSet<Integer> partitions) {
+    return kafkaStoreIngestionService.getConsumptionSnapshots(topicName, partitions);
+  }
+
+  @Override
+  public TopicPartitionIngestionContextResponse getTopicPartitionIngestionContext(
+      String versionTopic,
+      String topicName,
+      int partitionNum) {
+    return kafkaStoreIngestionService.getTopicPartitionIngestionContext(versionTopic, topicName, partitionNum);
+  }
+
+  @Override
+  public TopicPartitionIngestionContextResponse getHeartbeatLag(
```

Comment:
Maybe here we need to rename the Response class: `TopicPartitionIngestionContextResponse` a bit, it basically just wrap byte array for returning json string of server internal information.